docs/2.10.0/docs/upgrade/smart-contract-upgrades.rst

```diff
@@ -2008,6 +2011,152 @@ against those packages and its datatypes are not automatically converted.
+next SDK version. At this point, your package and any of its SCU upgrades would
+be stuck on that SDK version.
+
+For example, suppose you have a ``main`` package that depends on
+``daml-script-lts`` from SDK 2.10.0.
+
+.. code:: yaml
+
+  version: 2.10.0
+  name: main
+  ...
+  dependencies:
+  - daml-prim
+  - daml-stdlib
+  - daml-script-lts
+  build-options:
+  - --target=1.17
+
+.. code:: daml
+
+  module Main where
+
+  import qualified Daml.Script
+
+  data MyData = MyData
+    { field1 : Daml.Script.PartyIdHint
+    }
+
+Because ``MyData`` is a serializable datatype, any changes to it must be valid
+upgrade changes (e.g. adding a field) for the ``main`` package itself to be
+validly upgraded. If SDK 2.10.X were to introduce a change to
+``daml-script-lts`` that is not a valid upgrade of the ``daml-script-lts`` in
+SDK 2.10.0, then ``field1`` will not be upgradeable to the next version of the
+SDK, nor can the field be dropped because the field is used in SCU checks.
+
+At that point, all future development on ``main``, including future upgrades,
+would be locked to SDK 2.10.0. To bump the SDK version, ``main`` would have to
+be migrated via a manual upgrade tool with downtime -- existing downtime upgrade
+techniques are listed :ref:`here <upgrades-index>`.
+
+For this reason, we strongly recommend against using Daml Script as an upgrade
+dependency for any package going on the ledger. Whenever building the ``main``
+package above, you can expect a warning:
+
+.. code:: bash
+
+  > daml build
+  ...
+  warning while type checking data type Main.MyData:
+    This package depends on a datatype <package ID>:Daml.Script.Stable:PartyIdHint from <package ID> (daml-script-lts-stable, 0.0.0) with LF version 1.17.
+    
+    It is not recommended that >= LF1.17 packages use datatypes from Daml Script, because those datatypes will not be upgradeable.
+    Upgrade this warning to an error -Werror=upgrade-serialized-daml-script
+    Disable this warning entirely with -Wno-upgrade-serialized-daml-script
+
+**Note:** For added safety, you may upgrade these warnings to errors with
+``-Werror=upgrade-serialized-daml-script``. We recommend against disabling these
+warnings with ``-Wno-upgrade-serialized-daml-script``.
+
+If instead ``main`` depended on a datatype in a non-serializable position such
+as the type signature of a function, ``main`` could still be upgraded without
+breaking any SCU restrictions. For example:
+
+.. code:: daml
+
+  module Main where
+
+  import qualified Daml.Script
+
+  data MyData = MyData
+    { field1 : Text
+    }
+
+  myFunction : Daml.Script.PartyIdHint -> Bool
+  myFunction _ = True
+
+In this case, when changing SDK from 2.10.0 to 2.10.X, the typechecker will
+ignore the change of ``Daml.Script.PartyIdHint`` in ``myFunction``, because it
+is not in a serializable position. This means ``daml-script-lts`` can be kept
+even though when it is not a valid upgrade from one version to the next.
```

Comment:
Note that the upload check would still warn in this case though. We warn if daml-script is uploaded at all.
I realise we discussed this before in the context of depending on 1.15 packages.
Its okay to depend on 1.15 if not in a serializable position
Its less okay to depend on daml-script, even if not serializable, if its uploaded.
We should probably just add a caveat about this. It might need rethinking to be a nicer experience.

Reply:
I would say that having a 1.15 type in your template is worse than having a daml-script type though, since we are being careful with script. Up to you if you want to convey that